catalog/opensearch/catalog-opensearch-source/src/main/java/org/codice/ddf/opensearch/source/OpenSearchSource.java

```diff
@@ -1043,50 +1054,45 @@ public PointRadius getPointRadius() {
   @Nullable
   protected SpatialSearch createCombinedSpatialSearch(
       final Queue<PointRadius> pointRadiusSearches, final Queue<Geometry> geometrySearches) {
-    final boolean filterContainedSomePointRadiusCriteria =
-        CollectionUtils.isNotEmpty(pointRadiusSearches);
-    final boolean filterContainedSomePolygonSearchCriteria =
-        CollectionUtils.isNotEmpty(geometrySearches);
-
-    if (filterContainedSomePointRadiusCriteria && filterContainedSomePolygonSearchCriteria) {
-      LOGGER.debug(
-          "Ignoring all spatial searches because combining spatial and point radius searches is not yet implemented.");
-      return null;
-    }
-
-    if (!filterContainedSomePointRadiusCriteria && filterContainedSomePolygonSearchCriteria) {
-      if (CollectionUtils.size(geometrySearches) > 1) {
-        LOGGER.debug(
-            "Ignoring all polygon searches because combining polygon searches is not yet implemented.");
-        return null;
+    final boolean hasPointRadiusSearch = CollectionUtils.isNotEmpty(pointRadiusSearches);
+    final boolean hasGeometrySearches = CollectionUtils.isNotEmpty(geometrySearches);
+
+    Geometry geometrySearch = null;
+    BoundingBox boundingBox = null;
+    PointRadius pointRadius = null;
+    SpatialSearch spatialSearch = null;
+
+    if (hasGeometrySearches) {
+      if (geometrySearches.size() == 1) {
+        geometrySearch = geometrySearches.remove();
+      } else {
+        geometrySearch =
+            GEOMETRY_FACTORY.createGeometryCollection(geometrySearches.toArray(new Geometry[0]));
       }
-
-      final Geometry geometry = geometrySearches.remove();
-      if (!(geometry instanceof Polygon)) {
-        LOGGER.debug(
-            "Ignoring geometry search because only polygon searches are currently supported.");
-        return null;
+      if (geometrySearch instanceof Polygon && shouldConvertToBBox) {
+        boundingBox = BoundingBoxUtils.createBoundingBox((Polygon) geometrySearch);
+        geometrySearch = null;
       }
-
-      final Polygon polygon = (Polygon) geometry;
-      return shouldConvertToBBox
-          ? new SpatialSearch(BoundingBoxUtils.createBoundingBox(polygon))
-          : new SpatialSearch(polygon);
     }
 
-    if (filterContainedSomePointRadiusCriteria) {
-      if (CollectionUtils.size(pointRadiusSearches) > 1) {
-        LOGGER.debug(
-            "Ignoring all spatial searches because combining point radius-searches is not yet implemented.");
-        return null;
+    if (hasPointRadiusSearch) {
```

Comment:
Should at least always be able to convert to a bbox envelope for all geospatial constraints.  In the case of multiple point/radius without converting to bbox overall, could convert the individual point radius to their envelope and add them as polygons to the geometry collection to have an approximation

Reply:
Yes, making update to this logic shortly